.github/workflows/update.yaml

```diff
@@ -0,0 +1,106 @@
+name: Update Go Versions
+
+on:
+  schedule:
+    - cron: '0 22 * * 0-4' # 7:00 JST weekdays
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  update-go-versions:
+    runs-on: ubuntu-24.04
+    env:
+      GH_TOKEN: ${{ secrets.CYBOZU_NECO_PAT }}
+      UBUNTU_VERSIONS: "22.04 24.04"
+      GO_MAJOR_VERSIONS: "1.22 1.23"
+    
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Check for Go version updates
+        id: check-updates
+        run: |
+          NEED_UPDATE=0
+          touch ./update-body.txt
+          
+          # Get latest stable Go versions
+          STABLE_VERSIONS=$(curl -sSLf https://go.dev/dl/?mode=json | jq -r '[.[] | select(.stable == true)][0,1] | .version | sub("^go"; "")')
+          if [ -z "$STABLE_VERSIONS" ]; then
+            echo "Failed to fetch stable Go versions"
+            exit 1
+          fi
+
+          # Check if there are newer major versions available
+          HIGHEST_MAJOR_VERSION=$(echo "$STABLE_VERSIONS" | cut -d. -f1-2 | sort -V | tail -n1)
+          echo "HIGHEST_MAJOR_VERSION="$HIGHEST_MAJOR_VERSION
+          HIGHEST_CONFIGURED_VERSION=$(echo "${{ env.GO_MAJOR_VERSIONS }}" | tr ' ' '\n' | sort -V | tail -n1)
+          echo "HIGHEST_CONFIGURED_VERSION="$HIGHEST_CONFIGURED_VERSION
+          if [ "$(echo -e "$HIGHEST_MAJOR_VERSION\n$HIGHEST_CONFIGURED_VERSION" | sort -V | tail -n1)" = "$HIGHEST_MAJOR_VERSION" ] && [ "$HIGHEST_MAJOR_VERSION" != "$HIGHEST_CONFIGURED_VERSION" ]; then
```

Comment:
I feel this code for checking newer versions is complex. To make it simpler, how about detecting only the differences between the stable versions of Go and the versions we specify in `GO_MAJOR_VERSIONS` ? For example, we could use the `diff` command and write something like this:

```shell
GO_MAJOR_VERSIONS="1.23 1.22"
GO_STABLE_MAJOR_VERSIONS="1.24 1.23"
diff <(printf "%s\n" ${GO_MAJOR_VERSIONS}) <(printf "%s\n" ${GO_STABLE_MAJOR_VERSIONS})
```